### PR TITLE
FontAwesome 5.6.3 and updated versions of frameworks

### LINKED ISSUE #13849: Font Awesome Icons for MODX3 [Feature]
It would be awesome if MODX moved over to the new Font Awesome icon library, instead of using the old 4.7 library. Is there any plan for this?

### PR DESCRIPTION
**Pull request by alroniks**
### What does it do?
This PR provides updated tools for building default MODX manager theme to avoid various warnings messages during the build and also provides updated icons font [FontAwesome](https://fontawesome.com/) to the latest free version - 5.2.0 (**1295** icons against only **675** in 4.7)

### Why is it needed?
This PR solves issues with outdated packages that were used for building the default theme, also allows using new classes `.fa .fab .fas .far` for icons in the manager.

### Related issue(s)/PR(s)
Fixes #13849
#12790
#14030

All works fine, and old style of defining icons works also, but testing strongly needed.

### CODE REVIEW

**Review by Mark-H — DISMISSED** [review]

**Comment by ilyautkin** [comment]
I tested this PR. New icons work. No bugs found

1ynbwo

File: _build/templates/default/package.json [hunk 1]
```diff
@@ -1,25 +1,25 @@
 {
   "name": "revolution-theme-default",
   "title": "MODX Revolution",
-  "version": "2.3.0",
+  "version": "2.7.0",
```
Comment (Jako) [inline _build/templates/default/package.json]: 3.0.0?

File: _build/templates/default/package.json [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
     "grunt-contrib-uglify": "^0.9.2",
+    "grunt-contrib-watch": "^0.6.1",
     "grunt-growl": "^0.1.5",
     "grunt-imageoptim": "^1.4.1",
     "grunt-rename": "^0.1.4",
     "grunt-sass": "^1.0.0"
   },
   "dependencies": {
-    "node-sass": "^4.8.3"
+    "node-sass": "^4.9.2"
```
Comment (Jako) [inline _build/templates/default/package.json]: Should that be in devDependencies, since we use it for development only?

**Review by Jako — COMMENTED** [review]
Just two notes …

File: _build/templates/default/package.json (hunk 1 above)
Comment (alroniks) [reply _build/templates/default/package.json]: Indeed. First version was prepared for 2.7

**Comment by jonleverrier** [comment]
Just tested this PR. New icons work well!

The only small things I noticed whilst testing was, under elements tab;

- Some of the icons seem bolder than usual
- Some of the icons seem larger than before

shot1


In the new sidebar - some of the icons are now solid colour vs the outlined version before.

shot2

**Comment by alroniks** [comment]
It's why I wrote that testing needed, but it's awaited. The whole font was changed.

**Comment by jonleverrier** [comment]
What's the best way to provide you feedback @Alroniks ? I could make some small adjustments to the icons so they looked like they did before (as best as possible), whilst using the new Font Awesome.

**Comment by alroniks** [comment]
@jonleverrier You can push adjustments directly to this PR. I guess it's the best way.

**Comment by jonleverrier** [comment]
@Alroniks I've got a couple of commits left to push for the sidebar icons. I've tried pushing those commits to `git push Alroniks 3x-feature-13849_fa_and_updated_libs` after checking out with `git checkout --track Alroniks/3x-feature-13849_fa_and_updated_libs` but i get the following error in my console:

> ERROR: Permission to Alroniks/revolution.git denied to jonleverrier.
> fatal: Could not read from remote repository.

Does anybody know how I can resolve?

**Comment by jonleverrier** [comment]
Final bit of feedback - this PR seems to remove/break the arrows from the breadcrumb on resource pages. Strangely, it looks CSS related.

Before (3.x branch):
screen shot 2018-08-16 at 14 54 04

After (3x-feature-13849_fa_and_updated_libs branch):
screen shot 2018-08-16 at 14 54 18

**Comment by jonleverrier** [comment]
And strangely, when you shrink your browser window down to "mobile size", the mobile display no longer kicks in.

screen shot 2018-08-16 at 15 08 03

And upon refresh, you get a full screen of blue:

screen shot 2018-08-16 at 15 09 05

**Comment by alroniks** [comment]
It seems it should be updated to 5.3.0 already :) 
And bugs with the mobile view should be fixed, of course. If somebody can help with bourbon, it would be nice.

**Comment by Jako** [comment]
The build part is changed different in #14091 (removed bower). This could be added here too.

**Review by Mark-H — DISMISSED** [review]
Has a minor merge conflict but looks good

**Review by JoshuaLuckers — DISMISSED** [review]

**Comment by Mark-H** [comment]
I tried to merge and resolve the conflicts, but the removal of bower in #14091 and this are conflicting heavily. Afraid this will need some more work... do you think you have time to pick it up again @Alroniks?

**Comment by alroniks** [comment]
@Mark-H As FA released already 5.6 version it makes sense to close this PR and create a new. I will do it.

**Comment by alroniks** [comment]
@Mark-H I've updated font awesome to 5.6.3 and fixed conflicts. Now it should work.

**Review by Jako — CHANGES_REQUESTED** [review]
The code still uses bower which was removed in #14091. I could help to remove that part, if you want.

Or am I wrong with that?

**Review by Jako — APPROVED** [review]

**Review by Mark-H — APPROVED** [review]

**Pull request merged**